linetools/utils.py

```diff
@@ -346,84 +352,134 @@ def z_from_v(z, v):
     znew = (1+z)/R - 1
 
     return znew.value
+    """
+    raise DeprecationWarning("This function is deprecated, instead please use either z_from_dv() or dz_from_dv()"
+                             " depending on your needs.")
+
 
 # Slightly different functions for passing from dv to dz, and viceversa (that NT prefers).
 # May need to agree on one kind of conversion in the future
-def give_dv(z, zmean, rel=True):
-    """Gives velocity difference between z and zmean,
-    at zmean.
+def give_dz(dv, zref, rel=True):
+    """Same as dz_from_dv. This function will be deprecated."""
+    DeprecationWarning("This function will be deprecated. Please use instead dz_from_dv().")
+    return dz_from_dv(dv, zref, rel=rel)
+
+
+def give_dv(z, zref, rel=True):
+    """Same as dv_from_dz. This function will be deprecated."""
+    warnings.warn("This function will be deprecated. Please use instead dv_from_z().")
+    return dv_from_z(z, zref, rel=rel)
+
+
+def dv_from_z(z, zref, rel=True):
+    """Gives the rest-frame velocity difference dv
+    between z and zref. dv=0 at zref by definition.
 
     Parameters
     ---------
     z : float or np.array
         Redshifts to calculate dv on
-    zmean : float or np.array
-        Rest-frame redshift to perform the calculation.
-        If shape of zmean is equal than shape of z,
-        each dv is calculated at each zmean, otherwise zmean
-        is expected to be float
+    zref : float or np.array
+        Reference redshift where dv=0 by definition.
+        If shape of zref is equal than shape of z,
+        each dv is calculated at each zref, otherwise zref
+        es expected to be float.
     rel : bool, optional
         Whether to apply relativistic correction for
         a locally flat space-time. Default is True.
 
     Returns
     -------
     dv : Quantity or Quantity array
-        Rest frame velocity difference between z and zmean, at
-        zmean. It has same shape as z.
+        Rest-frame velocity difference between z and zref (dv=0 at zref by definition).
+        It has same shape as z.
         """
     z = np.array(z)
-    zmean = np.array(zmean)
+    zref = np.array(zref)
```

Comment:
You'll want to add checks as regards the rules you have in the docs
for the shapes of z and zref.